networkit/cpp/generators/test/GeneratorsGTest.cpp

```diff
@@ -228,6 +229,65 @@ TEST_F(GeneratorsGTest, testClusteredRandomGraphGeneratorSparseCommunities) {
     }
 }
 
+TEST_F(GeneratorsGTest, testConfigurationModelGeneratorWithEdgeSwitchingOnRealSequence) {
+    METISGraphReader reader;
+    auto graphs = {"input/jazz.graph", "input/lesmis.graph"};
+
+    for (auto path : graphs) {
+        Graph G = reader.read(path);
+        count n = G.numberOfNodes();
+        std::vector<count> sequence(n);
+        G.forNodes([&](node u) { sequence[u] = G.degree(u); });
+
+        bool skipTest = false;
+        EdgeSwitchingMarkovChainGenerator gen(sequence, skipTest);
+        Graph G2 = gen.generate();
+
+        count volume = std::accumulate(sequence.begin(), sequence.end(), 0);
+        EXPECT_EQ(volume, 2 * G2.numberOfEdges());
+
+        if (volume < 50000) {
+            std::vector<count> testSequence(n);
+            G2.forNodes([&](node u) { testSequence[u] = G2.degree(u); });
+            Aux::Parallel::sort(testSequence.begin(), testSequence.end(), std::greater<count>());
+            Aux::Parallel::sort(sequence.begin(), sequence.end(), std::greater<count>());
+
+            for (index i = 0; i < n; ++i) {
+                EXPECT_EQ(sequence[i], testSequence[i]);
+            }
+        }
```

Comment:
If this test fails, it's not clear for which instance it failed. I'd suggest splitting this test into two tests, one where the volume is expected to be < 50000 and another one where the volume is expected to be >= 50000.